### PR TITLE
i3645 finish new numismatics form

### LINKED ISSUE #3645: Refactor numismatics advanced search using the built-in advanced search BL form
The built-in advanced search form is included in BL 7 and BL 8.
Things to consider:
1. The queries will turn from q1, q2, q3 to [clause0], [clause1], [clause2] 
2. It uses view components rather the view partials. 
3. It will need a lot of work to make the form look as it does now.
4. It uses the json query DSL. This is the new way BL communicates with SOLR. This will require changes to the catalog controller.

### Acceptance criteria
- [ ] There is a new form
- [x] It is behind a feature flipper
- [x] It is ok if this form does not actually successfully perform a search - the search is blocked by changes in Solr / Blacklight

related to #3412

### PR DESCRIPTION
**Pull request by sandbergja**
Closes #3645 

To review, make sure that the "View components numismatics" feature is enabled

### CODE REVIEW

File: app/assets/stylesheets/components/search--advanced.scss [hunk 1]
```diff
@@ -120,3 +120,13 @@
 .row .range_limit input.range_begin, .row .range_limit input.range_end {
   width: 5em;
 }
+
```
Comment (sandbergja) [inline app/assets/stylesheets/components/search--advanced.scss:123]: Removing and improving some inline CSS from my previous PR

File: app/components/numismatics_search_form_component.html.erb [hunk 2]
```diff
@@ -37,4 +37,16 @@
...
+      <div class="form-group sort-buttons pull-left">
+        <%= content_tag :label, t('blacklight.advanced_search.form.sort_label'), id: 'advanced-search-sort-label', for: 'sort', class: 'control-label' %>
+        <%= sort_fields_select %>
+      </div>
+    <% end %>
+    <div class="submit-buttons pull-right form-group">
+      <%= submit_tag t('blacklight.advanced_search.form.search_btn_html'), class: 'btn btn-primary advanced-search-submit', id: "advanced-search-submit" %>
+    </div>
+
+  </div>
```
Comment (sandbergja) [inline app/components/numismatics_search_form_component.html.erb:51]: Adding the two elements that were missing from the form previously

File: spec/components/numismatics_search_form_component_spec.rb [hunk 3]
```diff
@@ -33,13 +33,31 @@
     allow(view_context).to receive(:facet_limit_for).and_return(nil)
   end
 
-  it "renders fields in the correct order" do
+  it "renders all expected fields" do
```
Comment (sandbergja) [inline spec/components/numismatics_search_form_component_spec.rb:36]: This "it" didn't accurately describe what was being tested

**Review by christinach — COMMENTED** [review]
@sandbergja thank you for your work on the blacklight upgrade! 🍄 
When I tested the numismatics form the one bug I can report, but it is not consistent between the fields is that when I tried to edit a search with no results it gave me the '404 something went wrong page'. [honeybadger error](https://app.honeybadger.io/projects/54399/faults/99540992)

File: app/models/search_builder.rb [hunk 4]
```diff
@@ -66,6 +66,7 @@ def facets_for_advanced_search_form(solr_p)
     # search form, so that the form displays accurate counts for
     # them in its dropdowns
     advanced_search_facets = blacklight_config.advanced_search.form_solr_parameters['facet.field']
+    solr_p[:fq]&.compact!
```
Comment (christinach) [inline app/models/search_builder.rb:69]: Thanks @sandbergja 🍄 !

**Review by christinach — APPROVED** [review]
Thanks 🍄 ! @sandbergja

**Pull request merged**